test/WebJobs.Script.Tests.Integration/TestScripts/CSharp/ProxyHttpRequest/run.csx

```diff
@@ -0,0 +1,10 @@
+
+using System.Net;
+using System.Net.Http.Headers;
+
+public static async Task<HttpRequestMessage> Run(HttpRequestMessage req, TraceWriter log)
+{
+    req.Headers.Add("ServerDateTime", DateTime.Now.ToString()); ;
```

Comment:
Also, wondering why the req needs to be returned, rather than modified in place. Is this to enable someone completely newing up a new request and returning it. Would that even work (would it mess up your req properties communication mechanism?)